systems/controllers/QPCommon.cpp

```diff
@@ -356,16 +356,30 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
   Vector6d body_vdot;
 
   for (int i=0; i < qp_input->num_tracked_bodies; i++) {
-    int body_id0 = qp_input->body_motion_data[i].body_id - 1;
-    double weight = params->body_motion[body_id0].weight;
-    desired_body_accelerations[i].body_id0 = body_id0;
+    if (qp_input->body_motion_data[i].body_id == 0)
+      mexErrMsgTxt("Body motion data with body id 0\n");
+    int body_or_frame_id0 = qp_input->body_motion_data[i].body_id - 1;
+    int true_body_id0 = body_or_frame_id0;
+    if (body_or_frame_id0 < 0){
+      // and get actual body using that; get back to the original
+      // frame id (undo the -1), and transform into index based
+      // on frame # by adding 1 and inverting
+      int frame_ind = -(body_or_frame_id0+2);
+      if (frame_ind >= pdata->r->num_frames)
+        mexErrMsgTxt("Got a frame ind greater than available!\n");
+      true_body_id0 = pdata->r->frames[frame_ind].body_ind;
+    }
```

Comment:
Should probably use RigidBodyManipulator::parseBodyOrFrameID (and make that method public).